src/parser.cpp

```diff
@@ -130,7 +138,7 @@ namespace ada {
           }
           // Otherwise, if remaining starts with an U+002F (/), set state to path or authority state
           // and increase pointer by 1.
-          else if (pointer + 1 < input.end() && pointer[1] == '/') {
+          else if (pointer + 1 < pointer_end && pointer[1] == '/') {
```

Comment:
same deal: `std::distance(pointer, pointer_end) < 1` or maybe `pointer != pointer_end` (would be faster if safe).